src/modules/kpi-reporting/controller.js

```diff
@@ -15,9 +15,8 @@ const getErrorMessage = (data, errorMessage) => {
  * @returns {Object} an object containing all the data sets required for the KPI UI
  */
 const getKPIData = async (request) => {
-  // get the last 2 returns cycles parameters
-  const returnCycles = await helpers.returns.date.createReturnCycles().slice(-2);
-  const [registrations, delegatedAccess, returnsDataMonthly, returnsDataCycle1, returnsDataCycle2, licenceNamesData] = await Promise.all([
+  const returnCycles = await getReturnCycles();
+  const [registrationsData, delegatedAccessData, returnsDataMonthly, returnsDataCycle1, returnsDataCycle2, licenceNamesData] = await Promise.all([
```

Comment:
Would it be better to use `Promise.allSettled` here? This would allow one or more errors, but with all of the taks being tried. E.g. 3 out of 6 could fail leaving 3 resuts to display.

Using Promise.all would mean that the first failure causings the full set to be rejected.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled